### PR TITLE
Document some local semgrep-agent use cases

### PR DESCRIPTION
**Pull request by underyx**

### CODE REVIEW

File: docs/integrations.md [hunk 1]
```diff
@@ -186,6 +186,48 @@ repos:
...
+```
+git checkout $RECENT_SHA
+docker run -v $(pwd):/src --workdir /src returntocorp/semgrep-agent:v1 python -m semgrep_agent --config p/r2c-ci --baseline-ref $OLDER_SHA
+```
+
+!!! info
+    The above commands all require `docker`
+    to be installed on your machine.
+    They also use Docker volumes
+    to make your working directory accessible to the container.`--config p/r2c-ci` sets the rules to be scanned,
```
Comment (ievans) [inline docs/integrations.md:227]:
```suggestion
    to make your working directory accessible to the container.`--config p/r2c-ci` semgrep rule configuration,
```

**Review by ievans — APPROVED** [review]
Awesome!

File: docs/integrations.md [hunk 2]
```diff
@@ -186,6 +186,49 @@ repos:
...
 Semgrep provides integrations with 3rd party services like Slack, Jira, Defect Dojo, and others. To configure these and learn more, visit [Manage > Notifications](https://semgrep.dev/manage/notifications).
 
+# Commit history
+
+While Semgrep CI is designed
+for integrating with various CI providers,
+it's versatile enough to be used locally
+to scan a repository with awareness of its git history.
+
+To locally scan what issues your current branch has
```
Comment (pabloest) [inline docs/integrations.md:196]:
```suggestion
To locally scan issues in your current branch
```

File: docs/integrations.md [hunk 3]
```diff
@@ -186,6 +186,49 @@ repos:
...
+Another use case is when you want to scan only commits
+from the past weeks for new issues they introduced.
+This can be done by using a git command
+that gets the tip of the current branch two weeks earlier:
+
+```
+docker run -v $(pwd):/src --workdir /src returntocorp/semgrep-agent:v1 python -m semgrep_agent --config p/r2c-ci --baseline-ref $(git rev-parse '@{2.weeks.ago}')
+```
+
+If you want to compare two commits
```
Comment (pabloest) [inline docs/integrations.md:213]:
```suggestion
To compare two commits
```

File: docs/integrations.md [hunk 4]
```diff
@@ -186,6 +186,49 @@ repos:
...
+and find the issues added between them,
+you need to checkout the more recent commit of the two
```
Comment (pabloest) [inline docs/integrations.md:215]:
```suggestion
checkout the more recent commit of the two
```

**Review by pabloest — COMMENTED** [review]
Added some minor copy suggestions but overall lgtm!

**Pull request merged**